topology/pynest/tests/test_spatial_kernels.py

```diff
@@ -0,0 +1,779 @@
+            L            : Side length of area / volume.
+            N            : Number of nodes.
+            kernel_name  : Name of kernel to use.
+            kernel_params: Dict with params to update.
+            open_bc      : Network with open boundary conditions
+            x0, y0       : Location of source neuron; open_bc only
+
+        Note
+        ----
+        For each new kernel to be added, the following needs to be
+        defined:
+
+        self._<kernel>  : lambda-function implementing kernel function
+        kernels entry   : mapping kernel name to kernel function
+        default_params  : default set of parameters for kernel
+        '''
+
+        if dim != 2 and open_bc:
+            raise ValueError("open_bc only supported for 2D")
+
+        self._seed = seed
+        self._dimensions = dim
+        self._L = float(L)
+        self._N = N
+        self._open_bc = open_bc
+        self._x_d, self._y_d = x0, y0
+
+        if (self._dimensions == 2):
+            if (self._open_bc):
+                self._max_dist = self._L * math.sqrt(2)
+            else:
+                self._max_dist = self._L / math.sqrt(2)
+        elif (self._dimensions == 3):
+            self._max_dist = self._L * math.sqrt(3) / 2
+
+        self._target_dists = None
+        self._all_dists = None
+
+        # kernel functions
+        self._constant = lambda D: self._params
+        self._linear = lambda D: self._params['c'] + self._params['a'] * D
+        self._exponential = lambda D: (self._params['c'] +
+                                       self._params['a'] *
+                                       math.exp(-D / self._params['tau']))
+        self._gauss = lambda D: (self._params['c'] +
+                                 self._params['p_center'] *
+                                 math.exp(-(D - self._params['mean']) ** 2 /
+                                        (2. * self._params['sigma'] ** 2)))
+        self._gamma = lambda D: (D ** (self._params['kappa'] - 1) /
+                                 (self._params['theta'] **
+                                  self._params['kappa'] *
+                                  scipy.special.gamma(self._params['kappa'])) *
+                                 math.exp(-D / self._params['theta']))
+
+        kernels = {
+            'constant': self._constant,
+            'linear': self._linear,
+            'exponential': self._exponential,
+            'gaussian': self._gauss,
+            'gamma': self._gamma}
+        self._kernel = kernels[kernel_name]
+
+        default_params = {
+            'constant': 1.,
+            'linear': {'a': -math.sqrt(2) / self._L, 'c': 1.0},
+            'exponential': {'a': 1.0, 'c': 0.0, 'tau': -self._L /
+                            (math.sqrt(2) * math.log((.1 - 0) / 1))},
+            'gaussian': {'p_center': 1., 'sigma': self._L / 4.,
+                         'mean': 0., 'c': 0.},
+            'gamma': {'kappa': 3., 'theta': self._L / 4.}}
```

Comment:
I tested it, an adequate parameter set for gamma would be 
` 'gamma': {'kappa': 4., 'theta': self._L / 20.}}`

Reply:
In which sense adequate? Wouldn't essentially any parameter set with kappa != 1 be useful?

Reply:
You are right, it is an personal view.